rmw_implementation/CMakeLists.txt

```diff
@@ -52,18 +52,37 @@ else()
 
   add_library(${PROJECT_NAME} SHARED
     src/functions.cpp)
+  target_include_directories(${PROJECT_NAME} PUBLIC
+    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+    "$<INSTALL_INTERFACE:include>")
```

Comment:
I have to say that I'm a bit concerned about making this header file "public".  If we put this in, then it seems like `load_library` and `lookup_symbol` are APIs that downstream consumers should be able to call.  And as far as I can tell, that is not the case.

One alternative here is to do something similar to what rviz_default_plugins does.  That is, put the header file in the src directory, and don't install it.  Then have the tests include the header file via relative paths.  I don't love that scheme, but it has the property that it keeps this implementation private (as it looks like it should be), but also makes the symbols available to tests.  What do you think?

Reply:
I don't have a strong opinion in favor or against exposing `rmw` implementation lookup. 

> One alternative here is to do something similar to what rviz_default_plugins does.

I'm OK with that scheme too.

Reply:
Though that doesn't change the fact that the symbols will be public.

Reply:
> Though that doesn't change the fact that the symbols will be public.

Just to recap a private conversation: while that is true, it is more around the messaging to external consumers.  If a function is in a publicly-accessible, installed header, then that seems like something a user might want to call.  If it is hidden away in a `src` directory, and the user has to go looking through symbols to find it, then it is pretty clear they are doing something unsupported.

Reply:
Agreed. See https://github.com/ros2/rmw_implementation/pull/135/commits/b885711253cbf7573593da68888d4f03f98858ef. Went the extra mile to only make symbols public if building for testing.